pandas/core/arrays/integer.py

```diff
@@ -581,6 +582,12 @@ def _maybe_mask_result(self, result, mask, other, op_name: str):
             result[mask] = np.nan
             return result
 
+        if result.dtype == "timedelta64[ns]":
```

Comment:
maybe also make this a free function as you did below in numpy_?

Reply:
this is in _maybe_mask_result, which is pretty similar thematically